app/controllers/actions/IdentifierAction.scala

```diff
@@ -38,14 +39,23 @@ class AuthenticatedIdentifierAction @Inject()(
                                              )
                                              (implicit val executionContext: ExecutionContext) extends IdentifierAction with AuthorisedFunctions {
 
+  private val enrolmentKey: String = "DAC6"
+
   override def invokeBlock[A](request: Request[A], block: IdentifierRequest[A] => Future[Result]): Future[Result] = {
 
     implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromHeadersAndSession(request.headers, Some(request.session))
 
-    authorised().retrieve(Retrievals.internalId) {
-      _.map {
-        internalId => block(IdentifierRequest(request, internalId))
-      }.getOrElse(throw new UnauthorizedException("Unable to retrieve internal Id"))
+    authorised().retrieve(Retrievals.internalId and Retrievals.authorisedEnrolments) {
```

Comment:
I think you need to authorise against the enrolment like you had before:
`authorised(Enrolment("DAC6")).retrieve(Retrievals.internalId and Retrievals.authorisedEnrolments)`

Reply:
Auth throws an AuthorisationException so I removed it for now.